hack/test_errata.py

```diff
@@ -0,0 +1,598 @@
+        {
+            "additional_necessary_info": "shouldn't be proccesed",
+            "msg": {
+                "errata_id": 5,
+                "product": "RHOSE",
+                "to": "SHIPPED_LIVE",
+            }
+        }
+    ]
+    expected_messages = [
+       {
+            "errata_id": 1,
+            "product": "RHOSE",
+            "to": "SHIPPED_LIVE",
+        },
+        {
+            "errata_id": 5,
+            "product": "RHOSE",
+            "to": "SHIPPED_LIVE",
+        }
+    ]
+    json_load_mock.return_value = {
+        "raw_messages": raw_messages,
+        "pages": 1
+    }
+    # Retrive messages from errata.poll
+    poll_period = datetime.timedelta(seconds=3600)
+    polled_messages = []
+    for message in errata.poll(period=poll_period):
+        polled_messages.append(message)
+    # Test if data was polled correctly
+    assert expected_messages == polled_messages
+
+
+@patch("builtins.print")
+@patch("urllib.request.urlopen")
+def test_notify_no_webhook(urlopen_mock, print_mock):
+    input_msgs = [
+        {
+            "errata_id": 1,
+            "product": "RHOSE",
+            "to": "SHIPPED_LIVE",
+        },
+        {
+            "errata_id": 5,
+            "product": "RHOSE",
+            "to": "SHIPPED_LIVE",
+        }
+    ]
+    errata.notify(message=input_msgs)
+    assert print_mock.call_args == (unittest.mock.call(input_msgs))
+
+    input_msgs = [
+        {
+            "errata_id": 3,
+            "product": "RHEL",
+            "to": "SHIPPED_LIVE",
+        },
+    ]
+    errata.notify(message=input_msgs)
+    assert print_mock.call_args == (unittest.mock.call(input_msgs))
+
+    input_msgs = [
+        {
+            "errata_id": 4,
+            "product": "RHEL",
+            "to": "QE",
+        },
+    ]
+    errata.notify(message=input_msgs)
+    assert print_mock.call_args == (unittest.mock.call(input_msgs))
+
+
+@patch("urllib.request.urlopen")
+def test_notify_format_of_message_pr_not_approved(urlopen_mock):
+    message = {
+        "fulladvisory": "RHEL",
+        "when": "today",
+        "synopsis": "OpenShift Container Platform X.X.X",
+        "uri": "example_uri"
+    }
+
+    errata.notify(message, True)
+
+    msg_text = '<!subteam^STE7S7ZU2>: {fulladvisory} shipped {when}: {synopsis} {uri}'.format(**message)
+    expected_data = urllib.parse.urlencode({
+        'payload': {
+            'text': msg_text,
+        },
+    }).encode('utf-8')
+    assert urlopen_mock.call_args[1]["data"] == expected_data
+
+
+@patch("urllib.request.urlopen")
+def test_notify_format_of_message_pr_approved(urlopen_mock):
+    message = {
+        "fulladvisory": "RHEL",
+        "when": "today",
+        "synopsis": "OpenShift Container Platform X.X.X",
+        "uri": "example_uri",
+        "approved_pr": "example_approved"
+    }
+
+    errata.notify(message, True)
+
+    msg_text = '<!subteam^STE7S7ZU2>: {fulladvisory} shipped {when}: {synopsis} {uri}'.format(**message)
```

Comment:
This needs to be a template set in `errata.py` which we'd reuse in unit tests

Reply:
That way we end up just duping the production logic in the test.  I'd rather just hard-code the expected string, instead of reformatting it from the `message`.